modules/p2/ding_serendipity_ting_entity/ding_serendipity_ting_entity.module

```diff
@@ -141,24 +145,26 @@ function ding_serendipity_ting_entity_ting_object_subject_serendipity_add($conte
   $obj = ding_entity_load($context['ting_object_id'], 'ting_object');
 
   // Build subject array.
-  $subject = $obj->getSubjects();
+  $subjects = $obj->getSubjects();
 
-  // Create the query.
-  foreach ($subject as & $facet) {
-    $facet = 'facet.subject="' . $facet . '"';
+  // Bail out if we don't have any subjects.
+  if (empty($subjects)) {
+    return $results;
   }
 
-  // Search with only subjects.
-  $query = implode(' OR ', $subject);
-  if (!empty($query)) {
-    $results = ding_serendipity_do_search($query, array());
-    ding_serendipity_exclude($results, $context['ting_object_id']);
+  // Find related content from any of the subjects.
+  $field_filters = array_map(function($subject) {
+    return new TingSearchFieldFilter(TingSearchCommonFields::SUBJECT, $subject, '=', BooleanStatementInterface::OP_OR);
+  }, $subjects);
 
-    foreach ($results as & $item) {
-      $item['info'] = array(
-        'description' => t('This material has been selected based on the subjects of the material you are viewing.'),
-      );
-    }
+  $query = ting_start_query()->addFieldFilters($field_filters);
+  $results = ding_serendipity_do_search($query);
+  ding_serendipity_exclude($results, $context['ting_object_id']);
```

Comment:
Wouldn't an extra `AND NOT rec.id =` be smarter than this?

Reply:
I'll add a note and fix it in the future.